Sources/ApplicativeRouter/ApplicativeRouter.swift

```diff
@@ -52,6 +52,22 @@ extension Parser {
   }
 }
 
+// MARK: - Monad
+
+extension Parser {
+  public func flatMap<B>( _ f: @escaping (A) -> Parser<I, B>) -> Parser<I, B> {
```

Comment:
one of the things I wanted to explore needed `flatMap` for parsers, so adding it. will explain more below...